File: backend/account/src/main/java/com/ssafy/account/api/response/account/HomeAccountResponse.java
package com.ssafy.account.api.response.account;

import com.ssafy.account.api.response.transaction.HomeTransactionResponse;
import com.ssafy.account.db.entity.account.Account;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class HomeAccountResponse { // 홈 화면의 계좌 정보를 나타내기 위한 DTO
    private Long accountId;
    private String accountName;
    private String accountNumber;
    private Long balance;
    private List<HomeTransactionResponse> transactions; // 최근 5개 거래목록

    public HomeAccountResponse(Account account, List<HomeTransactionResponse> transactions) {
        this.accountId = account.getId();
        this.accountName = account.getAccountName();
        this.accountNumber = account.getAccountNumber();
        this.balance = account.getBalance();
        this.transactions = new ArrayList<>(transactions);
    }
}

File: frontend/refacoring_end_file/app/src/main/java/com/allforyou/app/HomeAccountResponse.kt
package com.allforyou.app

data class HomeAccountResponse (
    val message: String,
    val code: String,
    val data : List<HomeAccount>? = null
){
    data class HomeAccount(
        val accountId : Long,
        val accountName : String,
        val accountNumber : String,
        var balance : Long,
        val transactions : List<HomeTransaction>
    )
}
